plugins/awards-backend/src/awards.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright SeatGeek
+ * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
+ */
+import { NotFoundError } from '@backstage/errors';
+import { Award, AwardInput } from '@seatgeek/backstage-plugin-awards-common';
+import { Logger } from 'winston';
+import { AwardsStore } from './database/awards';
+
+export class Awards {
+  private readonly db: AwardsStore;
+  private readonly logger: Logger;
+
+  constructor(db: AwardsStore, logger: Logger) {
+    this.db = db;
+    this.logger = logger.child({ class: 'Awards' });
+    this.logger.debug('Constructed');
+  }
+
+  async get(uid: string): Promise<Award> {
+    return await this.getAwardByUid(uid);
+  }
+
+  async create(input: AwardInput): Promise<Award> {
+    return await this.db.add(
+      input.name,
+      input.description,
+      input.image,
+      input.owners,
+      input.recipients,
+    );
+  }
+
+  async update(
+    identityRef: string,
+    uid: string,
+    input: AwardInput,
+  ): Promise<Award> {
+    const award = await this.getAwardByUid(uid);
+
+    if (!award.owners.includes(identityRef)) {
```

Comment:
is it guaranteed that owners will never be undefined?

Reply:
good question. as far as typescript is concerned, yes, but not sure if that could be missed on the db client underneath. 

i consider this PR one of many refactors needed to polish the originally hackathon code here

Reply:
It looks like the db store ensures that `undefined` values are converted to empty arrays to satisfy the TS interface

https://github.com/seatgeek/backstage-plugins/blob/197dae390cdb7ede426edc79433ca732a8d70509/plugins/awards-backend/src/database/awards.ts#L127